sql-plugin/src/main/scala/org/apache/spark/sql/rapids/RapidsShuffleInternalManagerBase.scala

```diff
@@ -313,10 +315,19 @@ abstract class RapidsShuffleInternalManagerBase(conf: SparkConf, val isDriver: B
       dependency: ShuffleDependency[K, V, C]): ShuffleHandle = {
     // Always register with the wrapped handler so we can write to it ourselves if needed
     val orig = wrapped.registerShuffle(shuffleId, dependency)
-    if (!shouldFallThroughOnEverything && dependency.isInstanceOf[GpuShuffleDependency[K, V, C]]) {
-      val handle = new GpuShuffleHandle(orig,
+    var shouldUseRapidsShuffle = false
+
+    if (!shouldFallThroughOnEverything) {
+      shouldUseRapidsShuffle = dependency match {
+        case gpuDependency: GpuShuffleDependency[K, V, C] =>
+          gpuDependency.useRapidsShuffle
+        case _ => false
+      }
+    }
```

Comment:
Can keep this a val by placing the boolean check in the match directly like this:
```suggestion
    val shouldUseRapidsShuffle = dependency match {
      case _ if shouldFallThroughOnEverything => false
      case gpuDependency: GpuShuffleDependency[K, V, C] =>
        gpuDependency.useRapidsShuffle
      case _ => false
    }
```